### PR TITLE
upgrade to eslint 8

### PR DESCRIPTION
**Pull request by ledhed2222**
upgrades @xrplf/eslint-config to use eslint 8 and cleans up both peer dependencies of @xrplf/eslint-config and dev dependencies of the whole project

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,10 +14,6 @@
     "test": "lerna run test"
   },
   "engines": {
-    "npm": ">= 7.0.0"
-  },
-  "dependencies": {
-    "@xrplf/eslint-config": "file:packages/eslint-config",
-    "@xrplf/prettier-config": "file:packages/prettier-config"
+    "npm": ">= 8.0.0"
```
Comment (ckniffen) [inline package.json:17]: I dont think our apps are ready to update to require 8

Comment (ckniffen) [reply package.json:17]: Also i dont think nom should ever be specified this way.

Comment (ledhed2222) [reply package.json:17]: this is just to develop this repo

**Comment by ckniffen** [comment]
When running this in xrpl.js it throws this error.  This can be recreated by running `npm pack` in `packages/eslint-config` and installing it into xrpl.js via a relative path ex. `npm i ../typescript-style/packages/eslint-config/xrplf-eslint-config-1.8.0.tgz`

```
> lint
> lerna run lint --stream
lerna notice cli v4.0.0
lerna info versioning independent
lerna info Executing command in 4 packages: "npm run lint"
ripple-address-codec: > ripple-address-codec@4.2.4 lint
ripple-address-codec: > eslint . --ext .ts
ripple-address-codec: Oops! Something went wrong! :(
ripple-address-codec: ESLint: 7.32.0
ripple-address-codec: Error: Failed to load plugin 'consistent-default-export-name' declared in '.eslintrc.js » @xrplf/eslint-config/base » /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@xrplf/eslint-config/rules/consistent-default-export-name.js': Cannot find module 'pkg-dir'
ripple-address-codec: Require stack:
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/common/resolve.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/rules/default-import-match-filename.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/index.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@eslint/eslintrc/lib/config-array-factory.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@eslint/eslintrc/lib/index.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/cli-engine/cli-engine.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/eslint/eslint.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/eslint/index.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/cli.js
ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/bin/eslint.js
ripple-address-codec: Referenced from: /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@xrplf/eslint-config/rules/consistent-default-export-name.js
ripple-address-codec:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
ripple-address-codec:     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
ripple-address-codec:     at Module.require (internal/modules/cjs/loader.js:974:19)
ripple-address-codec:     at require (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
ripple-address-codec:     at Object.<anonymous> (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/common/resolve.js:4:16)
ripple-address-codec:     at Module._compile (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
ripple-address-codec:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
ripple-address-codec:     at Module.load (internal/modules/cjs/loader.js:950:32)
ripple-address-codec:     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
ripple-address-codec:     at Module.require (internal/modules/cjs/loader.js:974:19)
ripple-address-codec: npm ERR! Lifecycle script `lint` failed with error:
ripple-address-codec: npm ERR! Error: command failed
ripple-address-codec: npm ERR!   in workspace: ripple-address-codec@4.2.4
ripple-address-codec: npm ERR!   at location: /Users/ckniffen/projects/ripple/xrpl.js/packages/ripple-address-codec
lerna ERR! npm run lint exited 1 in 'ripple-address-codec'
```

**Comment by ledhed2222** [comment]
> When running this in xrpl.js it throws this error. This can be recreated by running `npm pack` in `packages/eslint-config` and installing it into xrpl.js via a relative path ex. `npm i ../typescript-style/packages/eslint-config/xrplf-eslint-config-1.8.0.tgz`
> 
> ```
> > lint
> > lerna run lint --stream
> lerna notice cli v4.0.0
> lerna info versioning independent
> lerna info Executing command in 4 packages: "npm run lint"
> ripple-address-codec: > ripple-address-codec@4.2.4 lint
> ripple-address-codec: > eslint . --ext .ts
> ripple-address-codec: Oops! Something went wrong! :(
> ripple-address-codec: ESLint: 7.32.0
> ripple-address-codec: Error: Failed to load plugin 'consistent-default-export-name' declared in '.eslintrc.js » @xrplf/eslint-config/base » /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@xrplf/eslint-config/rules/consistent-default-export-name.js': Cannot find module 'pkg-dir'
> ripple-address-codec: Require stack:
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/common/resolve.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/rules/default-import-match-filename.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/index.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@eslint/eslintrc/lib/config-array-factory.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@eslint/eslintrc/lib/index.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/cli-engine/cli-engine.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/eslint/eslint.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/eslint/index.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/lib/cli.js
> ripple-address-codec: - /Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint/bin/eslint.js
> ripple-address-codec: Referenced from: /Users/ckniffen/projects/ripple/xrpl.js/node_modules/@xrplf/eslint-config/rules/consistent-default-export-name.js
> ripple-address-codec:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
> ripple-address-codec:     at Function.Module._load (internal/modules/cjs/loader.js:746:27)
> ripple-address-codec:     at Module.require (internal/modules/cjs/loader.js:974:19)
> ripple-address-codec:     at require (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
> ripple-address-codec:     at Object.<anonymous> (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/eslint-plugin-consistent-default-export-name/lib/common/resolve.js:4:16)
> ripple-address-codec:     at Module._compile (/Users/ckniffen/projects/ripple/xrpl.js/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
> ripple-address-codec:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
> ripple-address-codec:     at Module.load (internal/modules/cjs/loader.js:950:32)
> ripple-address-codec:     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
> ripple-address-codec:     at Module.require (internal/modules/cjs/loader.js:974:19)
> ripple-address-codec: npm ERR! Lifecycle script `lint` failed with error:
> ripple-address-codec: npm ERR! Error: command failed
> ripple-address-codec: npm ERR!   in workspace: ripple-address-codec@4.2.4
> ripple-address-codec: npm ERR!   at location: /Users/ckniffen/projects/ripple/xrpl.js/packages/ripple-address-codec
> lerna ERR! npm run lint exited 1 in 'ripple-address-codec'
> ```

removed that plugin, since we had already discussed getting rid of it. works for me now with xrpl.js

**Review by ckniffen — APPROVED** [review]

**Pull request merged**